src/Mod/Spreadsheet/TestSpreadsheet.py

```diff
@@ -1448,6 +1448,21 @@ def testParensAroundCondition(self):
         self.assertEqual(sheet.getContents('A1'), '=1 == 1 ? 1 : 0')
         self.assertEqual(sheet.A1, 1)
 
+    def testIssue6395(self):
+        """ Testing strings are correctly saved and restored"""
+        sheet = self.doc.addObject('Spreadsheet::Sheet','Spreadsheet')
+        sheet.set('A1', '\'36C') # Use a string that may be parsed as a quantity
+        self.doc.recompute()
+
+        self.doc.saveAs(self.TempPath + os.sep + 'string.fcstd')
+        FreeCAD.closeDocument(self.doc.Name)
+
+        self.doc = FreeCAD.openDocument(self.TempPath + os.sep + 'string.fcstd')
```

Comment:
For this case I don't think save and restore is really required, in master this test would fail even if you skip this.

Reply:
You asked for a test versus #6395

Reply:
Yes, but while #6395 is more obviously noticeable after save an restore the root of the problem happens immediately, the leading ' is not stored when it should, but testing save and restore also probably doesn't hurt.
I guess `sheet.get('A1')` would give a different result than before after save+restore but the test would already fail in the previous line.

It might be a good idea to add other potential failures to the test too, like `''asd` and `'1/2m` in case someone later on wants to bring back auto removal of leading ' but only takes the 36C exception into account.

Reply:
Well might be. Too tired with idea ATM. Will eventually come back to this later on.

Reply:
Adding more tests would certainly be nice: I propose that we merge this PR as-is (CONTRIBUTING.md does not require 100% test coverage) and that those interested in the topic continue development as needed.

Reply:
> I propose that we merge this PR

Beside tests, be aware that this PR will display all fixed string with a prepended single quote in the 'content' field.
To me it's acceptable, but YMMV.

Reply:
It's not my favorite, but I think for the time being it's worth it to resolve the bug. Once it's been in people's hands for a while we will see if it's too irksome and we need an alternative or refinement to it.

Reply:
I think it's better than the bug, we can try to improve later if there are complaints